### PR TITLE
SAK-29808 New User Activity Webservices

### PR DESCRIPTION
**Pull request by ddelblanco**
New Activity web services: getUserActivityByType, getUserLogonActivity
and getUserActivity/getUserActivityStringDates

### CODE REVIEW

File: webservices/cxf/src/java/org/sakaiproject/webservices/Activity.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        try {
+            conn = sqlService.borrowConnection();
+
+            ps = conn.prepareStatement(query);
+            for (int i = 0; i < args.length; i++) {
+                if (args[i] instanceof String) {
+                    ps.setString(i + 1, (String) args[i]);
+                } else if (args[i] instanceof Date) {
+                    // select * from sakai_event where event_date between to_date('2001-12-12 12:12','YYYY-MM-DD HH24:MI') and to_date('2017-12-12 12:12','YYYY-MM-DD HH24:MI')
+                    if (sqlService.getVendor().equals("oracle")) {
```
Comment (jonespm) [inline webservices/cxf/src/java/org/sakaiproject/webservices/Activity.java]: It was noted that this condition here is the same

**Comment by ddelblanco** [comment]
Review line 147

**Comment by ddelblanco** [comment]
Review if we can avoid the oracle-mysql differences

**Comment by ddelblanco** [comment]
Removed the unneeded condition from line 147.

Change the oracle-mysql differences is complex and involve a lot of changes. If not a major problem I think it should remain in that way. there is a way removing the "hour and minutes" but these parameters are useful, so I prefer to have that.

**Comment by steveswinsburg** [comment]
Web services should not be calling a persistence layer directly. This is code that should be exposed in an API, that the webservice would then call.

**Comment by ddelblanco** [comment]
I've been searching and seems that the event API, 
/kernel/kernel-impl/src/main/java/org/sakaiproject/event/
Ive not found any API to consult events from another tool. At least all the tools that I've found that does that, use directly SQL to check the events, never the event API.  Maybe I'm wrong and there is a way, if someone knows I can try to redo the webservices using that API.  If not, the only way to have this service is to write the API for events to allow to search in the events table.  If so, where do you thing is the best place to do it?   Maybe create a new class in the org.sakaiproject.event?  

Something like org.sakaiproject.event.api.EventQueryService;  

And then create in that class all the interesting queries to retrieve the events for a user, etc...?

**Comment by steveswinsburg** [comment]
Yes the APIs will need to be written for this. EventService is probably an ok place.

**Pull request closed without merging**